boards/FSM/Src/App/App_FlowMeter.c

```diff
@@ -1,17 +1,28 @@
-#include <stddef.h>
-#include <assert.h>
+#include "App_CanTx.h"
 #include "App_FlowMeter.h"
 
+extern struct World *world;
+
 #define MAX_NUM_OF_FLOWMETERS 2
 
 struct FlowMeter
 {
     float (*get_flow_rate)(void);
+    void (*set_periodic_signal)(
+        struct FsmCanTxInterface *can_tx_interface,
+        float                     value);
+    struct FsmCanTxInterface *can_tx;
 };
 
-struct FlowMeter *App_FlowMeter_Create(float (*get_flow_rate)(void))
+struct FlowMeter *App_FlowMeter_Create(
+    struct FsmCanTxInterface *const can_tx,
```

Comment:
App_FlowMeter should not be coupled with CAN (i.e. `can_tx`, and `set_periodic_signal` should be removed entirely). Simply expose a getter to the frequency. The state machine will be responsible for using this getter and setting the CAN value there.

You can check out `App_DriveState_SetImdPeriodicCanSignals`  in #555 for an example

Reply:
The `can_tx` doesn't seem removed yet @pasitlaothamatas